pyproject.toml

```diff
@@ -0,0 +1,48 @@
+[tool.poetry]
+name = "csm-api-client"
+version = "1.0.0"
+description = "Python client library for CSM APIs"
+authors = [
+    "Ryan Haasken <ryan.haasken@hpe.com>",
+    "Eli Kamin <eli.kamin@hpe.com>",
+    "Jack Stanek <jack.stanek@hpe.com>",
+    "Pat Tovo <pat.tovo@hpe.com"
+]
+license = "MIT"
+readme = "README.md"
+homepage = "https://github.com/Cray-HPE/sat"
+packages = [{include = "csm_api_client"}]
+
+[tool.poetry.dependencies]
+python = "^3.8"
+boto3 = "^1.24.0"
+inflect = ">=0.2.5"
+kubernetes = "^23.0.0"
+PyYAML = "<6.0"
+requests = "<3.0"
+requests-oauthlib = "^1.3.1"
+cray-product-catalog = "^1.6.0"
+
+[tool.poetry.group.dev.dependencies]
+nose2 = {version = "^0.12.0", extras = ["coverage_plugin"]}
+mypy = "^0.971"
+types-PyYAML = "<6.0"
+types-oauthlib = "^3.2.0"
+types-requests = "<3.0"
+boto3-stubs = {version = "^1.24.0", extras = ["s3"]}
+
+[[tool.poetry.source]]
+name = "algol60"
+url = "https://artifactory.algol60.net/artifactory/csm-python-modules/simple/"
+default = false
+secondary = false
+
+[tool.mypy]
+disallow_untyped_defs = true
+ignore_missing_imports = true
+files = "csm_api_client"
+exclude = "^tests/.*"
+
+[build-system]
+requires = ["poetry-core>=1.2.0"]
```

Comment:
I think the latest version of `poetry-core` is actually 1.1.0 - 
```
mbp-ekamin:python-csm-api-client ekamin$ /tmp/poetryvenv/bin/pip freeze | grep poetry-core
poetry-core==1.1.0
```

This ends up causing a problem when trying to install the library with `pip`.

Reply:
`poetry` itself released version 1.2.0 while I was working on this ticket, and I wanted to try out some of its new features (e.g. plugins). I think `poetry-core` 1.2.0 is distributed with poetry through its installer but isn't available directly on pip. I think removing the version constraint is sufficient to resolve this.